configure

```diff
@@ -699,6 +699,18 @@ cat > ${foo}.f <<EOF
           end program test_exit
 EOF
 
+  # check if gfortran > 9
+  grep '^SFC' configure.wrf | grep -i 'gfortran' > /dev/null
+  if [ $? ]
+  then
+    if [ $(${SFC} -dumpversion) -gt 9 ]
```

Comment:
Where the SFC variable is set?

Reply:
This code also will fail to check the conditional

```shell
[wrf@cbe07e141d13 /]$ test $(gfortran -dumpversion) -gt 9
bash: test: 10.1.0: integer expression expected
[wrf@cbe07e141d13 /]$ test 10.1.0 -gt 9
bash: test: 10.1.0: integer expression expected
```

Reply:
Missed the fact that "-dumpversion" distro specific. Output of dumpversion depends on how the GCC is configured for the distro. Would n't work for all cases. 

Below patch should fix this.

`
diff --git a/configure b/configure
index 2d7b64c0..6966fdb9 100755
--- a/configure
+++ b/configure
@@ -685,6 +685,26 @@ if [ $os = "Linux" -o $os = "Darwin" ]; then

   foo=foo_$$

+  cat > ${foo}.F << EOF
+      PROGRAM GFORTRAN_VERSION_CHECK
+
+        IF (__GNUC__ .GT. 9) CALL EXIT(1)
+
+      END PROGRAM
+EOF
+
+  grep '^SFC' configure.wrf | grep -i 'gfortran' > /dev/null
+  if [ $? ]
+  then
+    gfortran -o ${foo} ${foo}.F > /dev/null 2>&1
+    ./${foo}
+    if [ $? ]
+    then
+      sed '/^FCBASEOPTS_NO_G/ s/$/ -fallow-argument-mismatch -fallow-invalid-boz/' configure.wrf > configure.wrf.edit
+      mv configure.wrf.edit configure.wrf
+    fi
+  fi
+
 cat > ${foo}.c <<EOF
  int main(int argc, char ** argv)
  {

`

Reply:
Missed the fact that output of dumpversion is distro specific. Output depends on configure options used to build GCC for the distro. Would not work is all cases.

Alternate fix could be

`diff --git a/configure b/configure`
`index 2d7b64c0..b2fb28fa 100755`
`--- a/configure`
`+++ b/configure`
`@@ -685,6 +685,25 @@ if [ $os = "Linux" -o $os = "Darwin" ]; then`
` `
`   foo=foo_$$`
` `
`+  cat > ${foo}.F << EOF`
`+      PROGRAM GFORTRAN_VERSION_CHECK`
`+`
`+        IF (__GNUC__ .GT. 9) CALL EXIT(1)`
`+`
`+      END PROGRAM`
`+EOF`
`+`
`+  grep '^SFC' configure.wrf | grep -i 'gfortran' > /dev/null`
`+  if [ $? ]`
`+  then`
`+    gfortran -o ${foo} ${foo}.F > /dev/null 2>&1 && ./${foo}`
`+    if [ $? ]`
`+    then`
`+      sed '/^FCBASEOPTS_NO_G/ s/$/ -fallow-argument-mismatch -fallow-invalid-boz/' configure.wrf > configure.wrf.edit`
`+      mv configure.wrf.edit configure.wrf`
`+    fi`
`+  fi`
`+`
` cat > ${foo}.c <<EOF `
`  int main(int argc, char ** argv)`
`  {`

Reply:
I cherry-picked the code, modified and applied to my code

This uses a subprocess to change the IFS and parse the string resulting from run `$(gfortran -dumpversion)`

This works in Archlinux.

```shell
grep '^SFC' configure.wrf | grep -i 'gfortran' > /dev/null
if [ $? ]; then
(
    IFS=. read major minor micro build <<< $(gfortran -dumpversion)
		 
    if [[ $major -gt 9 ]];then
	  echo "Patching config.wrf"
	  echo
	  sed '/^FCBASEOPTS_NO_G/ s/$/ -fallow-argument-mismatch -fallow-invalid-boz/' configure.wrf > configure.wrf.edit
          mv configure.wrf.edit configure.wrf
    fi
)
fi
```

At least I can confirm that the addiction of the flags resolves the problem